### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by jackheslop96**
- Updating XPath logic after edge case discovered in QA
- Ensuring submission status is set back to submitted after an amendment

### CODE REVIEW

**Review by vajarova — APPROVED** [review]

**Review by vajarova — DISMISSED** [review]

File: app/models/XPath.scala [hunk 1]
```diff
@@ -32,47 +32,46 @@ case class XPath(value: String) {
...
-      case x if x.matches("^Consignment/containerIndicator$")                                                              => TransportDetails
-      case x if x.matches("^Consignment/inlandModeOfTransport$")                                                           => TransportDetails
-      case x if x.matches("^Consignment/modeOfTransportAtTheBorder$")                                                      => TransportDetails
-      case x if x.matches("^Consignment/referenceNumberUCR$")                                                              => TransportDetails
-      case x if x.matches("^Consignment/CountryOfRoutingOfConsignment(.+)$")                                               => RouteDetails
-      case x if x.matches("^Consignment/TransportCharges(.+)$")                                                            => TransportDetails
-      case x if x.matches("^Consignment/HouseConsignment(\\[[\\d]*])/ConsignmentItem(.+)$")                                => Items
-
+    val pf: PartialFunction[String, Task] = {
+      case x if x.matches("^(.*)/TransitOperation/declarationType$")                                                            => PreTaskList
```
Comment (joecefnhedog) [inline app/models/XPath.scala:36]: is it (.*) at the beginning now - because you deleted the replace thing?

Comment (jackheslop96) [reply app/models/XPath.scala:36]: @joecefnhedog yep the problem was two-fold

The error pointer after an IE013 rejection would be something like `/CC013C/Consignment/...`

Also, we saw an error pointer in QA like `//Consignment/...`

This updated logic ensures we cover a much wider range of scenarios

Comment (joecefnhedog) [reply app/models/XPath.scala:36]: Looks safer to me 👍

File: app/models/XPath.scala [hunk 2]
```diff
@@ -32,47 +32,46 @@ case class XPath(value: String) {
...
 
   def task: Option[Task] = {
-    val pf: PartialFunction[String, Task] = _.replace("/CC015C/", "") match {
-      case x if x.matches("^TransitOperation/declarationType$")                                                            => PreTaskList
-      case x if x.matches("^TransitOperation/TIRCarnetNumber$")                                                            => PreTaskList
-      case x if x.matches("^TransitOperation/security$")                                                                   => PreTaskList
-      case x if x.matches("^TransitOperation/bindingItinerary$")                                                           => RouteDetails
-      case x if x.matches("^TransitOperation/reducedDatasetIndicator$")                                                    => TraderDetails
-      case x if x.matches("^TransitOperation/limitDate$")                                                                  => TransportDetails
-      case x if x.matches("^Authorisation(.+)$")                                                                           => TransportDetails
```
Comment (joecefnhedog) [inline app/models/XPath.scala:42]: and the .+ means one or more occurrence, so it wouldn't match on exactly "Authorisation" - looks good then

**Review by joecefnhedog — APPROVED** [review]

**Review by vajarova — APPROVED** [review]

**Pull request merged**